### PR TITLE
[Keyboard] Add keyboard Reviung61

### PR DESCRIPTION
**Pull request by gtips**
Add keyboard Reviung61.

## Description
Standard ANSI 60% keyboard. And compatible with MX and ALPS  

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/reviung61/info.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    "layouts": {
+        "LAYOUT": {
+            "layout": [
+                {"label":"k00", "x":0, "y":0},
+                {"label":"k01", "x":1, "y":0},
+                {"label":"k02", "x":2, "y":0},
+                {"label":"k10", "x":0, "y":1, "w":1.5},
+                {"label":"k12", "x":1.5, "y":1, "w":1.5}
+            ]
+        }
```
Comment (zvecr) [inline keyboards/reviung61/info.json:16]:
```suggestion
        "LAYOUT_60_ansi": {
            "layout": [
                {"x":0, "y":0},
                {"x":1, "y":0},
                {"x":2, "y":0},
                {"x":3, "y":0},
                {"x":4, "y":0},
                {"x":5, "y":0},
                {"x":6, "y":0},
                {"x":7, "y":0},
                {"x":8, "y":0},
                {"x":9, "y":0},
                {"x":10, "y":0},
                {"x":11, "y":0},
                {"x":12, "y":0},
                {"x":13, "y":0, "w":2},

                {"x":0, "y":1, "w":1.5},
                {"x":1.5, "y":1},
                {"x":2.5, "y":1},
                {"x":3.5, "y":1},
                {"x":4.5, "y":1},
                {"x":5.5, "y":1},
                {"x":6.5, "y":1},
                {"x":7.5, "y":1},
                {"x":8.5, "y":1},
                {"x":9.5, "y":1},
                {"x":10.5, "y":1},
                {"x":11.5, "y":1},
                {"x":12.5, "y":1},
                {"x":13.5, "y":1, "w":1.5},

                {"x":0, "y":2, "w":1.75},
                {"x":1.75, "y":2},
                {"x":2.75, "y":2},
                {"x":3.75, "y":2},
                {"x":4.75, "y":2},
                {"x":5.75, "y":2},
                {"x":6.75, "y":2},
                {"x":7.75, "y":2},
                {"x":8.75, "y":2},
                {"x":9.75, "y":2},
                {"x":10.75, "y":2},
                {"x":11.75, "y":2},
                {"x":12.75, "y":2, "w":2.25},

                {"x":0, "y":3, "w":2.25},
                {"x":2.25, "y":3},
                {"x":3.25, "y":3},
                {"x":4.25, "y":3},
                {"x":5.25, "y":3},
                {"x":6.25, "y":3},
                {"x":7.25, "y":3},
                {"x":8.25, "y":3},
                {"x":9.25, "y":3},
                {"x":10.25, "y":3},
                {"x":11.25, "y":3},
                {"x":12.25, "y":3, "w":2.75},

                {"x":0, "y":4, "w":1.25},
                {"x":1.25, "y":4, "w":1.25},
                {"x":2.5, "y":4, "w":1.25},
                {"x":3.75, "y":4, "w":6.25},
                {"x":10, "y":4, "w":1.25},
                {"x":11.25, "y":4, "w":1.25},
                {"x":12.5, "y":4, "w":1.25},
                {"x":13.75, "y":4, "w":1.25}
            ]
        }
```

File: keyboards/reviung61/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+// Defines names for use in layer keycodes and the keymap
+enum layer_names {
+  _BASE,
+  _FN
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  /* Base */
+  [_BASE] = LAYOUT_reviung61(
```
Comment (zvecr) [inline keyboards/reviung61/keymaps/default/keymap.c:26]:
```suggestion
  [_BASE] = LAYOUT_60_ansi(
```

File: keyboards/reviung61/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    KC_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,
+    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,
+    KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT, 
+    KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT,
+    KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, KC_RGUI, MO(_FN), KC_RCTL
+
+  ),
+  [_FN] = LAYOUT_reviung61(
```
Comment (zvecr) [inline keyboards/reviung61/keymaps/default/keymap.c:34]:
```suggestion
  [_FN] = LAYOUT_60_ansi(
```

File: keyboards/reviung61/reviung61.h [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+ * | 20   | 21| 22| 23| 24| 25| 26| 27| 28| 29| 2a| 2b|     2c |
+ * |-----------------------------------------------------------|
+ * | 30     | 31| 32| 33| 34| 35| 36| 37| 38| 39| 3a|       3b |
+ * |-----------------------------------------------------------|
+ * | 40 | 41 | 42 |        45              | 48 | 49 | 4a | 4b |
+ * `-----------------------------------------------------------'
+ */
+
+
+#define LAYOUT_reviung61( \
```
Comment (zvecr) [inline keyboards/reviung61/reviung61.h:45]:
```suggestion
#define LAYOUT_60_ansi( \
```

File: keyboards/reviung61/rules.mk [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```
Comment (zvecr) [inline keyboards/reviung61/rules.mk:32]:
```suggestion

LAYOUTS = 60_ansi
```

File: keyboards/reviung61/info.json [hunk 6]
```diff
@@ -0,0 +1,18 @@
+{
+    "keyboard_name": "reviung61",
+    "url": "",
+    "maintainer": "gtips",
+    "width": 3,
+    "height": 2,
```
Comment (zvecr) [inline keyboards/reviung61/info.json:6]:
```suggestion
    "width": 15,
    "height": 5,
```

File: keyboards/reviung61/rules.mk [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (fauxpark) [inline keyboards/reviung61/rules.mk:11]:
```suggestion
```

File: keyboards/reviung61/rules.mk [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
```
Comment (fauxpark) [inline keyboards/reviung61/rules.mk:31]:
```suggestion
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/reviung61/keymaps/default/rules.mk [hunk 9]
```diff
@@ -0,0 +1 @@
+# RGBLIGHT_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/reviung61/keymaps/default/rules.mk:1]: This file can be deleted.

File: keyboards/reviung61/keymaps/default/config.h [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+// place overrides here
+
+// --- if use RGB LED ---
```
Comment (fauxpark) [inline keyboards/reviung61/keymaps/default/config.h:21]: Maybe this should go in a separate `default_rgb` keymap?

File: keyboards/reviung61/config.h [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID 0xFEED
+#define PRODUCT_ID 0x0000
```
Comment (fauxpark) [inline keyboards/reviung61/config.h:24]:
```suggestion
#define PRODUCT_ID 0x7C1A
```
```
echo -n "reviung61" | sha1sum | cut -c1-4
```

File: keyboards/reviung61/info.json (hunk 6 above)
Comment (gtips) [reply keyboards/reviung61/info.json]: Thanks!

File: keyboards/reviung61/keymaps/default/config.h (hunk 10 above)
Comment (gtips) [reply keyboards/reviung61/keymaps/default/config.h]: I see, Thanks!

File: keyboards/reviung61/keymaps/default/config.h [hunk 12]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2020 gtips
```
Comment (zvecr) [inline keyboards/reviung61/keymaps/default/config.h:1]: This file can be removed

Comment (gtips) [reply keyboards/reviung61/keymaps/default/config.h]: Thanks. removed this file.

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**